src/Nox.Reference.Data.World/Extensions/WorldQueryExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Nox.Reference.Abstractions;
+
+namespace Nox.Reference.Data;
+
+public static class WorldQueryExtensions
+{
+    public static ICurrencyInfo Get(this IQueryable<ICurrencyInfo> query, string currency)
+    {
+        return query.First(x => x.IsoCode == currency);
```

Comment:
should be add auto-detect logic or TODO to add it here?

Reply:
Sure, it will be another task how to resolve key for certain entities.